core/src/main/java/org/apache/iceberg/StaticDataTask.java

```diff
@@ -71,8 +71,10 @@ public List<DeleteFile> deletes() {
 
   @Override
   public CloseableIterable<StructLike> rows() {
-    StructProjection projection = StructProjection.create(tableSchema, projectedSchema);
-    Iterable<StructLike> projectedRows = Iterables.transform(Arrays.asList(rows), projection::wrap);
+    Iterable<StructLike> projectedRows =
```

Comment:
Note that I changed this to always create a new `StructProjection`. Without this all the rows when reading metadata tables were identical since they kept using the same projection.

I don't believe this should be problematic since `StaticDataTask` is only used for metadata tables, but happy to hear any alternatives.

Reply:
This isnt great as it will have a perf hit on the non Generic use case.  Is there a reason it works now in Spark engine?

I havent checked it, but it may be more an issue about ManifestReader using Avro.read().reuseContainer() (ie, we need to make a defensive copy like in many places in the code)

Reply:
@szehon-ho I believe it work in Spark, because the Spark API [iterates one row at a time](https://github.com/apache/iceberg/blob/master/spark/v3.1/spark/src/main/java/org/apache/iceberg/spark/source/BaseDataReader.java#L103), using [this iterator](https://github.com/apache/iceberg/blob/master/spark/v3.1/spark/src/main/java/org/apache/iceberg/spark/source/RowDataReader.java#L87).

I tested changing `reuseContainer` in `ManifestReader`, but it does not work. I believe the problem is with always using the same `StructProjection`. One alternative I see is modifying the `DataTask` interface and add a `reuseContainer` method there that that Generic can then set to false but it will be set to true everywhere else. What do you think?